Started using `spherical_jn` instead of `sph_jn`

mamakancha opened the pull request:
This PR fixes #12369

mamakancha committed:
Started using `spherical_jn` instead of `sph_jn`

This PR fixes #12369

mamakancha commented:
I ran tests on `special`. Everything seems fine with the syntax I guess. Please review and suggest changes.

bjodah commented:
`spherical_jn` is quite new, we should support somewhat older versions of scipy when we can:
```
try:
    from scipy.special import spherical_jn
    f = lambda ...
except ImportError:
    from scipy.special import sph_jn
    f = lambda ...
```

mamakancha committed:
Some more changes made in `bessel.py`

Except block in `Line 960` has been updated and trailing whitespaces have been removed.

mamakancha commented:
Sorry. Had some problem submitting the PR. Requesting a new PR.
